app/frontend/components/domains/external-api-key/external-api-key-modal-sub-route.tsx

```diff
@@ -0,0 +1,247 @@
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Grid,
+  GridItem,
+  HStack,
+  IconButton,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react"
+import { Info, Key, Prohibit } from "@phosphor-icons/react"
+import { addYears } from "date-fns"
+import { observer } from "mobx-react-lite"
+import React, { useEffect, useState } from "react"
+import { FormProvider, useForm } from "react-hook-form"
+import { useTranslation } from "react-i18next"
+import { useNavigate, useParams } from "react-router-dom"
+import { IExternalApiKey } from "../../../models/external-api-key"
+import { useMst } from "../../../setup/root"
+import { IExternalApiKeyParams } from "../../../types/api-request"
+import { CopyableValue } from "../../shared/base/copyable-value"
+import { DatePickerFormControl, TextFormControl, UrlFormControl } from "../../shared/form/input-form-control"
+import { RemoveConfirmationModal } from "../../shared/remove-confirmation-modal"
+
+interface IProps {}
+
+interface IExternalApiKeyForm extends IExternalApiKeyParams {
+  revokedAt?: Date
+}
+
+const formFormDefaultValues = (externalApiKey?: IExternalApiKey): IExternalApiKeyForm => {
+  return {
+    name: externalApiKey?.name || "",
+    connectingApplication: externalApiKey?.connectingApplication || "",
+    expiredAt: externalApiKey?.expiredAt ?? addYears(new Date(), 2),
+    webhookUrl: externalApiKey?.webhookUrl,
+    revokedAt: externalApiKey?.revokedAt,
+  }
+}
+
+export const ExternalApiKeyModalSubRoute = observer(function ExternalApiKeyModalSubRoute() {
+  const { t } = useTranslation()
+  const { externalApiKeyId } = useParams()
+  const navigate = useNavigate()
+  const { jurisdictionStore } = useMst()
+  const { currentJurisdiction } = jurisdictionStore
```

Comment:
a better way to get the currentJurisdiction is with the useJurisdiction hook

Reply:
unless there is already a parent component that uses it

Reply:
@bkmaibach parent component uses it. This is a subroute that get's rendered in an `Outlet` . https://reactrouter.com/en/main/components/outlet